lib/services/recipes_service.dart

```diff
@@ -0,0 +1,171 @@
+import 'package:MobileOne/dao/recipes_dao.dart';
+import 'package:MobileOne/data/recipe.dart';
+import 'package:MobileOne/data/wishlist_item.dart';
+import 'package:MobileOne/services/user_service.dart';
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:flutter/material.dart';
+import 'package:get_it/get_it.dart';
+import 'package:uuid/uuid.dart';
+
+import 'image_service.dart';
+
+class RecipesService {
+  final RecipesDao dao = GetIt.I.get<RecipesDao>();
+  final UserService userService = GetIt.I.get<UserService>();
+
+  Map<String, dynamic> _recipes;
+  Map<String, List<WishlistItem>> _recipeItems = {};
+
+  void _flush() {
+    _recipes = null;
+  }
+
+  Map<String, dynamic> get recipes => _recipes;
+
+  List<Recipe> sortRecipes(List<Recipe> recipiesList) {
+    if (recipiesList != null) {
+      recipiesList.sort(
+          (a, b) => a.label.toLowerCase().compareTo(b.label.toLowerCase()));
+    }
+
+    return recipiesList;
+  }
+
+  Future<Map<String, dynamic>> fetchRecipes() async {
+    DocumentSnapshot snapshot = await dao.fetchRecipes(userService.user.uid);
+
+    final recipes = snapshot?.data();
+
+    _recipes = {};
+
+    if (recipes != null) {
+      List<Recipe> recipesList = recipes.entries.map((element) {
+        return Recipe.fromMap(element.key, element.value);
+      }).toList();
+
+      recipesList = sortRecipes(recipesList);
+
+      recipesList.forEach((element) {
```

Comment:
why do you make an inverted convertion from list of object to map ? keep list of object